### ISSUE TITLE
Improve FileStorage documentation

### ISSUE THREAD

**Issue by thewoz** [opencv/opencv#25219]
### Describe the doc issue

Hello everyone,
I wanted to try to use FileStorage to upload an XML file and couldn't do it. 
So I created a post on the OpenCV forum and one on Stackoverflow

https://forum.opencv.org/t/using-opencv-filestorage-to-read-a-xml-file/16812/5

https://stackoverflow.com/questions/78161034/using-opencv-filestorage-to-read-a-xml-file?noredirect=1#comment137802795_78161034

(Sorry for the crosspost)

What I didn't realize is that I can't just load any file.
My proposal is to explain the situation better on the documentation.
One could take a cue from @crackwitz 's answer 

### Fix suggestion

OpenCV does not read arbitrary XML/YAML/JSON. For that purpose you need to use a generic XML/YAML/JSON library.
cv::FileStorage is made specifically to store and read OpenCV types (e.g. cv::Mat). 
It assumes a specific structure to the file. This is specific to cv::FileStorage. For example in the case of a XML file one requirement is that all the data is contained in a <opencv_storage> tag.

**Comment by crackwitz** [comment]
Docs reside in `modules/core/include/opencv2/core/persistence.hpp`

Docs contain an XML example:

https://github.com/opencv/opencv/blob/625eebad54a34a7bdad6812f3e9ec050a1b3adc5/modules/core/include/opencv2/core/persistence.hpp#L65-L80

**Comment by thewoz** [comment]
What I'm suggesting, and what I think is very useful, that the same information was written on this page:

https://docs.opencv.org/3.4/da/d56/classcv_1_1FileStorage.html

**Comment by mshabunin** [comment]
Please check this tutorial: https://docs.opencv.org/4.9.0/dd/d74/tutorial_file_input_output_with_xml_yml.html
Does it contain necessary information?

**Comment by mshabunin** [comment]
I agree we need some sort of note regarding limits of XML/YAML/JSON support.

Also, there is another page with persistence intro: https://docs.opencv.org/4.9.0/d4/da4/group__core__xml.html Perhaps we could extract it as a separate markdown page and add links to it from the class documentation and tutorial.
